# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/jimmy/ne/i18n-charcase.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/edit.c.orig	2001-09-01 00:32:32.000000000 +0300
+++ ./src/edit.c	2003-06-21 16:08:51.000000000 +0300
@@ -51,7 +51,7 @@
 	while (b->cur_pos < b->cur_line_desc->line_len && isalpha((unsigned char)b->cur_line_desc->line[b->cur_pos])) {
 		c = b->cur_line_desc->line[b->cur_pos];
 		delete_char(b, b->cur_line_desc, b->cur_line, b->cur_pos);
-		insert_char(b, b->cur_line_desc, b->cur_line, b->cur_pos, to(c));
+		insert_char(b, b->cur_line_desc, b->cur_line, b->cur_pos, to((unsigned char) c));
 		char_right(b);
 	}
 
--- ./src/inputclass.c.orig	2001-09-01 00:32:32.000000000 +0300
+++ ./src/inputclass.c	2003-06-21 16:00:24.000000000 +0300
@@ -357,7 +357,7 @@
 /* This vector contains n in the nth entry, except for the lower
 case letters, where it contains n-32. */
 
-const unsigned char up_case[256] = {
+unsigned char up_case[256] = {
 	0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF,
 	0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F,
 	0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D, 0x2E, 0x2F,
--- ./src/ne.c.orig	2001-09-01 00:32:32.000000000 +0300
+++ ./src/ne.c	2003-06-21 15:58:54.000000000 +0300
@@ -107,8 +107,8 @@
 
 
 /* The main() function. It is responsible for argument parsing, calling
-some terminal and signal initialization functions, and entering the
-event loop. */
+some terminal, signal and vector initialization functions, and entering
+the event loop. */
 
 int main(int argc, char **argv) {
 
@@ -123,6 +123,9 @@
 
 	setlocale(LC_ALL, "");
 
+	/* Correct the up_case vector according to the current locale. */
+	for(c=128; c<256; c++) up_case[c] = toupper(c);
+
 	for(i=1; i<argc; i++) {
 
 		/* Special arguments start with two dashes. If we find one, we
--- ./src/ne.h.orig	2001-09-01 00:32:32.000000000 +0300
+++ ./src/ne.h	2003-06-21 16:01:54.000000000 +0300
@@ -436,7 +436,7 @@
 /* This vector is a translation table for the regex library which maps
 lower case characters to upper case characters. It is also used elsewhere. */
 
-extern const unsigned char up_case[256];
+extern unsigned char up_case[256];
 
 /* This number defines the macro hash table size. This table can have
 conflicts. */
